application/src/main/java/net/corda/v5/application/persistence/PagedQuery.java

```diff
@@ -68,5 +69,33 @@
          */
         @NotNull
         List<R> getResults();
+
+        /**
+         * Returns {@code true} if the {@link ResultSet} has more results to retrieve.
+         * <p>
+         * Modifying the {@link PagedQuery} linked to the {@link ResultSet} does not affect {@link ResultSet}'s
+         * behaviour when calling {@link ResultSet#next()}.
+         *
+         * @return {@code true} if the {@link ResultSet} has more results to retrieve.
+         *
+         * @return
+         */
+        boolean hasNext();
+
+        /**
+         * Executes the query linked to the {@link ResultSet} to retrieve the next page of results.
```

Comment:
I'm ok with this API change, but it could be confusing because we don't have any guarantees that the resultset doesn't change between result pages. This is one of the reasons I originally preferred the explicit offset/limit API which reflects what a DB would do.
This API could give the impression that the resultset is fixed when `execute()` is called and you page through those results, which is not the case.
So if we implement this API, I think we need to be very specific about the fact that these guarantees don't exist in the general docs but also i these JavaDocs I think.